src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax.strategy;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.NoneExistedOperationHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Operation;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class OperationStrategyImplTest {
+    private final OperationHandler balance = new BalanceOperationHandler();
+    private final OperationHandler supply = new SupplyOperationHandler();
+    private final OperationHandler purchase = new PurchaseOperationHandler();
+    private final OperationHandler returning = new ReturningOperationHandler();
+    private final FruitTransaction balanceLemon = new FruitTransaction(
+            Operation.BALANCE, "lemon", 2);
+    private final FruitTransaction supplyLemon = new FruitTransaction(
+            Operation.SUPPLY, "lemon", 2);
+    private final FruitTransaction purchaseLemon = new FruitTransaction(
+            Operation.PURCHASE, "lemon", 2);
+    private final FruitTransaction returnLemon = new FruitTransaction(
+            Operation.RETURN, "lemon", 2);
+    private final OperationHandler wrongHandler = new NoneExistedOperationHandler();
+    private final List<OperationHandler> wrongOperationList = List.of(
+            wrongHandler, wrongHandler, wrongHandler, wrongHandler);
+    private final List<OperationHandler> goodOperationList = List.of(
+            balance, supply, purchase, returning);
+    private final OperationStrategy actualStrategy = new OperationStrategyImpl(goodOperationList);
+    private List<OperationHandler> actual;
+
+    @Test
+    void operationStrategy_InputOk_BalanceTrue() {
+        actual = actualStrategy.getHandlers(balanceLemon);
+        assertEquals(1, actual.size());
+        assertEquals("BalanceOperationHandler", actual.get(0).getClass().getSimpleName());
+    }
```

Comment:
You can compare using getClass like this: assert(BalanceOperationHandler.class, actual.get(0).getClass());